core/src/main/java/org/apache/accumulo/core/fate/zookeeper/FateLock.java

```diff
@@ -184,17 +214,11 @@ public SortedMap<Long,Supplier<FateLockEntry>> getEarlierEntries(long entry) {
       }
 
       for (String name : children) {
-        // this try catch must be done inside the loop because some subset of the children may exist
-        try {
-          long order = Long.parseLong(name.substring(PREFIX.length()));
-          if (order <= entry) {
-            byte[] data = zoo.getData(path + "/" + name);
-            // Use a supplier so we don't need to deserialize unless the calling code cares about
-            // the value for that entry.
-            result.put(order, Suppliers.memoize(() -> FateLockEntry.deserialize(data)));
-          }
-        } catch (KeeperException.NoNodeException ex) {
-          // ignored
+        var parsed = new NodeName(name);
+        if (predicate.test(parsed.sequence, parsed.fateLockEntry)) {
+          // Use a supplier so we don't need to deserialize unless the calling code cares about
```

Comment:
You may want to move this comment to NodeName where the supplier is created now, but it doesn't matter too much either way probably